src/get_the_state_of_the_system.py

```diff
@@ -4,13 +4,19 @@
 
 # main function
 def get_the_state(token: str, server_id: str) -> str:
-    if not server_id:
-        return ''
-    # extract dscServerId
-    response = ea_rest_call(f'/api/dtools/DscServer/{server_id}', 'GET', token)
-    dsc_server_id = response["serverId"]
-    # use dscServerId to get the ApplicationStatus
-    response2 = ea_rest_call(f'/api/jobProcessing/DscServerApplicationStatus/dscserver/{dsc_server_id}', 'GET', token)
-    # extract state_information and change the JSON-string to a python-dictionary
-    state_information = json.loads(response2[0]['stateDetails'])
+    try:
+        if not server_id:
+            return ''
+        # extract dscServerId
+        response = ea_rest_call(f'/api/dtools/DscServer/{server_id}', 'GET', token)
+        dsc_server_id = response["serverId"]
+        # use dscServerId to get the ApplicationStatus
+        response2 = ea_rest_call(f'/api/jobProcessing/DscServerApplicationStatus/dscserver/{dsc_server_id}', 'GET', token)
+        # in case response is empty
+        if response2 is None:
```

Comment:
```suggestion
        if not response2:
```
When the system was offline an empty array was returned, not None. This checks for either empty array or None (or empty dict)